api/features/views.py

```diff
@@ -187,6 +189,53 @@ def get_serializer_context(self):
 
         return context
 
+    def apply_state_to_queryset(
+        self, query_data: dict[str, typing.Any], queryset: QuerySet[Feature]
+    ) -> QuerySet[Feature]:
+        if not query_data.get("environment"):
+            raise serializers.ValidationError(
+                "Environment is required in order to filter by state search or by state enabled"
+            )
+        state_enabled = query_data["state_enabled"]
+        state_search = query_data["state_search"]
+        environment_id = query_data["environment"]
+
+        filter_search_q = Q()
+        if state_search is not None:
+            filter_search_q = filter_search_q | Q(
+                feature_state_value__string_value__icontains=state_search,
+            )
+
+            if state_search.lower() in {"true", "false"}:
+                boolean_search = state_search.lower() == "true"
+                filter_search_q = filter_search_q | Q(
+                    feature_state_value__boolean_value=boolean_search
+                )
```

Comment:
This should be:

```suggestion
                filter_search_q = filter_search_q | Q(
                    feature_state_value__boolean_value=boolean_search,
                    feature_state_value__type=BOOLEAN
                )
```

(same goes for the string and integer filters below)